mlflow/langchain/__init__.py

```diff
@@ -899,37 +905,34 @@ def _load_model_from_local_fs(local_model_path, model_config_overrides=None):
         return _load_model(local_model_path, flavor_conf)
 
 
-class _LoadedModelTracker:
+class _ModelTracker:
     """
-    Tracks models loaded by `load_model`.
+    Tracks models existing in current context.
     """
 
     def __init__(self):
         self.model_ids: dict[int, str] = {}
-        # temporary solution to track model_id for autologging
-        # TODO: remove this and pass model_id to MlflowLangchainTracer
-        self._last_model_id = None
+        self._lock = threading.Lock()
+        self._model_locks = defaultdict(threading.Lock)
```

Comment:
curious why we need two types of locks.

Reply:
Oh the _lock is used for clear method. Probably we can remove this one since it shouldn't be used by users?